frontend/storage/models/GameSpoilers.ts

```diff
@@ -0,0 +1,23 @@
+export interface GameSpoilersProps{
```

Comment:
I can see where you're coming from :).

Okay, that's mostly a sensible approach given the existing requirements but if you take a step back (or maybe forward) you'll see that what you have here is probably not what you want. First a small exercise:

Somewhere down the line you'll add `Game` model and a list of games to the store. It'll likely be a map, mapping `ID` => `Game` for easy lookup. Then think about how you'll interact with the data. Let's say you want to implement the view that shows the list of games. Okay, initially it'll be something like: **a)** get all games, **b)** display them in the list. Fine by now. But you also need to make it work with the spoilers, you don't want to show games in the list that the user has not played, right?
Okay, so you take your array of games. `filter()` it and for each game check it against your `gameSpoilers`. At the moment you probably would do something like `filter(game => store.gameSpoilers.get(game.id).showSpoilers)`. 
Then similar solution for the ability page, hide the game-examples for hidden games.

There's a fair bit of redundant information in your model right now, and if you remove the redundancy you're left with a class that really all it does is stores a single boolean. Can you simplify it further?

Reply:
What I am getting at here is this: whether the user wants to see a game is a simple boolean. You don't need to store the game's name in there, because it's just an attachment of a boolean to a game. 

The simplest solution would be to just store the information whether the game is spoiler or not in `Game` model itself, but that's a bad idea because `isSpoiler` is a function of both a game and the user. It's not Game's responsibility ot know whether the user wants to see it. So we don't want to go in that direction.

But there is an alternative. If you have a list of Games and you want to know a certain boolean for those games, you can just store that information as `gameId` => `boolean` map. I understand why you decided to create this model the way you did it - you don't yet have a game model and the task was to connect config view to redux. The simplest way to go (which often is the best option) is to directly solve the problem, and you needed a way to display the game's name. 

But in this case that resulted in a data model that doesn't work with the later requirements. The lesson is to always try to think of a change in the broader context and look into the future (and often do some foreseeing and guessing).

Reply:
To break it down into actionable steps:
1. You need to create the `Game` model first, make it as minimal as you want, just ID and name will be fine.
2. Add a map of games to your store (mapped by ID)
3. Remove your `GameSpoiler` model altogether and store the information as a map of `gameId` => `boolean`
4. As for those mutators - scrap that. The reason I had them in Metroidvania Maps project in the first place was because it was a direct refactor of having **A LOT** of duplicate code. We likely won't have that kind of code here, so they only add confusion :).